tests/plugins/test_amp_plugins.py

```diff
@@ -72,6 +72,11 @@ def test_amp_apex_ddp(
 class GradientUnscaleBoringModel(BoringModel):
 
     def on_after_backward(self):
+        # TODO: replace with `on_before_optimizer_step` so we don't need to check accumulate and unscale manually
+        if self.trainer.fit_loop.should_accumulate():
+            return
+        opt = self.optimizers()
+        self.trainer.precision_plugin.scaler.unscale_(opt)
```

Comment:
When having `on_before_optimizer_step`. Let's make sure we don't unscaled twice.

Reply:
Yeah, will remember to update this in the followup

Reply:
Is this also reflected in the docs? Otherwise this is a BC on master.

Reply:
I don't think the docs mentioned the skip on accumulation or that the loss was unscaled before.

I'll add these docs improvements when `on_before_optimizer_step` is added.